src/OnlineSales/Data/ApiDbContext.cs

```diff
@@ -78,32 +79,61 @@ public ApiDbContext(DbContextOptions<ApiDbContext> options, IConfiguration confi
 
     public virtual DbSet<ChangeLogTaskLog>? ChangeLogTaskLog { get; set; }
 
-    public override Task<int> SaveChangesAsync(bool acceptAllChangesOnSuccess, CancellationToken cancellationToken = default)
+    public override async Task<int> SaveChangesAsync(bool acceptAllChangesOnSuccess, CancellationToken cancellationToken = default)
     {
+        List<ChangeLog> changes = new ();
+
         var entries = ChangeTracker
        .Entries()
        .Where(e => e.Entity is BaseEntity && (
                e.State == EntityState.Added
                || e.State == EntityState.Modified));
 
+        List<AuditEntry> auditEntries = new ();
+
         foreach (var entityEntry in entries)
         {
             if (entityEntry.State == EntityState.Modified)
-            {
+            { 
                 ((BaseEntity)entityEntry.Entity).UpdatedAt = DateTime.UtcNow;
                 ((BaseEntity)entityEntry.Entity).UpdatedByIp = httpContextHelper!.IpAddress;
                 ((BaseEntity)entityEntry.Entity).UpdatedByUserAgent = httpContextHelper!.UserAgent;
             }
-
-            if (entityEntry.State == EntityState.Added)
+            else if (entityEntry.State == EntityState.Added)
             {
                 ((BaseEntity)entityEntry.Entity).CreatedAt = DateTime.UtcNow;
                 ((BaseEntity)entityEntry.Entity).CreatedByIp = httpContextHelper!.IpAddress;
                 ((BaseEntity)entityEntry.Entity).CreatedByUserAgent = httpContextHelper!.UserAgent;
             }
+
+            auditEntries.Add(new AuditEntry()
+            {
+                EntityEntry = entityEntry,
+                EntityState = entityEntry.State,
+            }); 
         }
 
-        return base.SaveChangesAsync(acceptAllChangesOnSuccess, cancellationToken);
+        await base.SaveChangesAsync(acceptAllChangesOnSuccess, cancellationToken);
+
+        foreach (var entry in auditEntries)
+        {
+            if (entry.EntityState == EntityState.Added || entry.EntityState == EntityState.Modified)
```

Comment:
1) @WASandaruwan why do you need a filter here if you already filter it in where condition when you request entries from ChangeTracker?
2) This whole look could be replaces with an easy to ready LINQ expression which will convert auditEntries into change log entries.